CHANGELOG.md

```diff
@@ -1,3 +1,63 @@
+# 0.28 - 2021-04-05 "The Taproot Release"
+
+At nearly nine months, this is our longest release cycle ever, and thanks
+to a huge increase in the number of active contributors this year and last,
+it is also **by far** our largest release ever, at 148 PRs merged from 23
+different contributors. Our primary goal in this release was to introduce
+support for Taproot and its associated data structures: addresses, taptrees,
+sighashes, PSBT fields, and more. As it turned out, these changes required
+(or at least, incentivized) changing a lot of our APIs, causing a significant
+increase in scope.
+
+We have more big changes coming down the pike. 2022 is going to be a big
+year for `rust-bitcoin`, which we know is exciting for us but disruptive to
+downstream users who ultimately want the library to just wark. Our hope is
+that by 2023 we will have eliminated large amounts of technical debt,
+modernized our APIs to meet current Rust conventions, and clarified the scope
+of the individual crates in this ecosystem while still providing the essential
+functionality needed by our downstream users, especially wallet projects.
+
+We will also develop a plan to make our releases more predictable and manageable,
+likely by having scheduled releases with limited scope. We would like to reach
+a point where we no longer have frequent breaking releases, but right now we
+are nowhere close.
+
+Upcoming changes will include
+- A quick new release which updates our MRSV from 1.29 to 1.41 and does little else
```

Comment:
@Kixunil I know you would like to get some more breaking changes in related to Taproot and PSBT. Would it be ok if we bumped the MSRV first?

Reply:
I don't remember which Taproot/PSBT breaking changes I wanted but yes, I do think bumping MSRV should be one of the first tasks. That being said, I suggest we try to merge as many open PRs first to minimize the churn. I'd rather redo semi-automated edition fixes than rebase all other PRs.